pkg/cloudprovider/providers/azure/azure_loadbalancer.go

```diff
@@ -105,7 +127,50 @@ func (az *Cloud) EnsureLoadBalancer(clusterName string, service *v1.Service, nod
 		}
 	}
 
-	lb, lbNeedsUpdate, err := az.reconcileLoadBalancer(lb, pip, clusterName, service, nodes)
+	var lbIP *string
```

Comment:
Yeah, the function gets a bit long. But we'll also need to pass down the external parameters to the spited func.

I've got a try for the change: https://github.com/karataliu/kubernetes/commit/3848fb498be410ef84ac8347d495ba0571e3e511 , and could be evaluated later.